🔧 chore(Makefile): update SRC variable to include 'lists/init' file

LithiumOx opened the pull request:
📝 docs(enum.h): add enum for different types of tokens

🔧 fix(minishell.h): update last modified timestamp in file header
🔧 fix(minishell.h): update last modified timestamp in file header
🔧 fix(structs.h): update last modified timestamp in file header
✨ feat(minishell.h): add support for bool type
✨ feat(minishell.h): add support for unistd.h library
✨ feat(minishell.h): add support for list_append function
✨ feat(minishell.h): add support for list_insert function
✨ feat(structs.h): add support for t_token struct
✨ feat(structs.h): add support for t_token type field
✨ feat(structs.h): add support for t_token value field
✨ feat(structs.h): add support for t_token prev field
✨ feat(structs.h): add support for t_token next field

📝 docs(init.c): add documentation for list_append and list_insert functions

LithiumOx committed:
🔧 chore(Makefile): update SRC variable to include 'lists/init' file
📝 docs(enum.h): add enum for different types of tokens

🔧 fix(minishell.h): update last modified timestamp in file header
🔧 fix(minishell.h): update last modified timestamp in file header
🔧 fix(structs.h): update last modified timestamp in file header
✨ feat(minishell.h): add support for bool type
✨ feat(minishell.h): add support for unistd.h library
✨ feat(minishell.h): add support for list_append function
✨ feat(minishell.h): add support for list_insert function
✨ feat(structs.h): add support for t_token struct
✨ feat(structs.h): add support for t_token type field
✨ feat(structs.h): add support for t_token value field
✨ feat(structs.h): add support for t_token prev field
✨ feat(structs.h): add support for t_token next field

📝 docs(init.c): add documentation for list_append and list_insert functions

LithiumOx commented:
Deze branch hoef je nog niet te evalueren. Heb hier een begin gemaakt met de types linked lists en structs. Puur ter voorbereiding van de lexer en parser

LithiumOx committed:
🔀 chore(enum.h): update enum values and comments
🔀 chore(structs.h): update timestamp in comments

LithiumOx committed:
Merge branch 'check_bad_quotes_parantheses' into lists+enum+struct

LithiumOx committed:
🔧 chore(c_cpp_properties.json): update includePath for Mac configuration
🔧 chore(Makefile): update INCLUDES variable to include additional directories
🔧 chore(libft): update libft submodule commit reference
🔧 chore(src/lexer): add new files index.c and utils.c

LithiumOx committed:
🔧 fix(c_cpp_properties.json): update include paths to include subdirectories
🔧 fix(Makefile): remove lists/init from the list of source files
🔧 fix(includes/minishell.h): update timestamp
🔧 fix(includes/structs.h): update timestamp
🔧 fix(src/main.c): add tokenization logic

LithiumOx committed:
🔧 fix(minishell.h): update last modified timestamp
🔧 fix(libft): update libft submodule commit
🔧 fix(main.c): fix memory leaks and add debug print statements
✨ feat(main.c): add support for tokenizing input commands in minishell

LithiumOx committed:
🔧 chore(Makefile): fix typo in INCLUDES variable
🔧 chore(libft): update libft submodule to commit e940f84ca25690b4f7d9300ed027922dff3e5a4b
🔧 chore(main.c): refactor print_vec function to pretty_print_vector
🔧 chore(main.c): refactor clear_structs_vec function to clear_token
🔧 chore(main.c): update print_vec function to pretty print tokens
🔧 chore(main.c): update clear_structs_vec function to properly free memory

LithiumOx committed:
🐛 fix(libft): update subproject commit hash
🐛 fix(main.c): remove unnecessary initialization of tokens vector

LithiumOx committed:
🔀 chore(libft): update subproject commit hash
🐛 fix(main.c): fix memory leak by initializing tokens vector before each iteration

LithiumOx committed:
🐛 fix(main.c): remove unnecessary printf statement and free memory in tokenize function

LithiumOx committed:
📝 chore(launch.json): add launch configuration for debugging
🔄 chore(libft): update libft submodule to commit 7e3359b
🐛 fix(main.c): fix pretty_print_vector function formatting
🐛 fix(main.c): fix token type assignment in tokenize function
🐛 fix(main.c): remove check_quotes_parantheses function call in main function

LithiumOx committed:
🐛 fix(launch.json): update launch arguments to a longer string
🐛 fix(main.c): fix comparison function from mini_strcmp to ft_strcmp
✨ feat(main.c): improve pretty printing of tokens in the vector

LithiumOx committed:
🔄 chore(libft): update subproject commit hash

LithiumOx committed:
🔄 chore(libft): update subproject commit hash

LithiumOx committed:
🔄 chore(libft): update subproject commit hash

LithiumOx committed:
🐛 fix(main.c): fix typo in variable name 'lenght' to 'length' in pretty_print_vector function
🐛 fix(main.c): fix memory leak by freeing token in clear_token function
🐛 fix(main.c): fix incorrect assignment of token value in tokenize function
🐛 fix(main.c): fix incorrect push of token address in tokenize function

LithiumOx committed:
🔄 chore(libft): update subproject commit hash

LithiumOx committed:
🐛 fix(main.c): fix indentation in pretty_print_vector function

LithiumOx committed:
🔧 chore(launch.json): update args in Debug configuration
📦 chore(main.c): update timestamp in Updated comment
🔧 chore(print_vector.c): add new file print_vector.c

LithiumOx committed:
🔀 chore(libft): update libft submodule to commit 9f4b1373abd91ac57a2fa8785dd683d7c879b8d2
🐛 fix(main.c): fix token indexing in pretty_print_vector function
✨ feat(main.c): add capitalize_token function to convert token values to lowercase

LithiumOx committed:
🐛 fix(enum.h): update timestamp in file header
🐛 fix(main.c): change function name capitalize_token to uncapitalize_token
🐛 fix(main.c): change variable name tokens to t in tokenize function
🐛 fix(main.c): change variable name tokens to t in main function

LithiumOx committed:
🐛 fix(main.c): fix typo in variable name 't' to 'tokens'
✨ feat(main.c): add dynamic tokenization of input string

LithiumOx committed:
Merge branch 'main' into lists+enum+struct

LithiumOx committed:
🔀 chore(structs.h): update include statement for enum.h
🔄 refactor(structs.h): update timestamp in file header comment

JdeBaaij commented:
Bij input:

./minishell "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."

Krijg je een memory leak(detecteerbaar met leaks -q minishell)
de memory leak geeft het adres aan van token0 -> value

Echter wanneer ik een losse free toevoeg voor token 0 dan zegt het programma dat ik dubbel free dus honestly idfk???


Ik had ter controle dit toegevoegd aan tokenize en toen geprobeerd free(ptoken) maar dat werkte dus niet....

         t_token	*ptoken;
	ptoken = (t_token *)ft_vec_get(&tokens, 0);
	printf("│ptoken ------------ Adress: %p\n", ptoken);



leaks Report Version: 4.0
Process 93515: 177 nodes malloced for 15 KB
Process 93515: 1 leak for 1536 total leaked bytes.

    1 (1.50K) ROOT LEAK: 0x7fbbcd008800 [1536]
    
    
    ptoken ------------ Adress: 0x7fbbcd008800

Im a little lost ngl brother

LithiumOx committed:
🔄 chore(libft): update libft submodule to commit 0a4dba3e649d20d5d6d57d19a022943d8e63c6fc

LithiumOx committed:
🔧 chore(Makefile): add DEBUG variable and DEBUG_FLAGS for conditional debugging
🔧 chore(minishell.h): set DEBUG to 0 if not defined
🔧 chore(main.c): add check_leaks function and call it at exit if DEBUG is enabled

Commit:
lexer pre test

Commit:
module update
